x-pack/test/functional/services/cases/common.ts

```diff
@@ -0,0 +1,152 @@
+
+      // case description
+      const descriptionArea = await find.byCssSelector('textarea.euiMarkdownEditorTextArea');
+      await descriptionArea.focus();
+      await descriptionArea.type('Test description');
+
+      // save
+      await testSubjects.click('create-case-submit');
+
+      await testSubjects.existOrFail('case-view-title');
+    },
+
+    /**
+     * Goes to the first case listed on the table.
+     *
+     * This will fail if the table doesn't have any case
+     */
+    async goToFirstListedCase() {
+      await testSubjects.existOrFail('cases-table');
+      await testSubjects.click('case-details-link');
+      await testSubjects.existOrFail('case-view-title');
+    },
+
+    /**
+     * Marks a case in progress via the status dropdown
+     */
+    async markCaseInProgressViaDropdown() {
+      await this.openCaseSetStatusDropdown();
+
+      await testSubjects.click('case-view-status-dropdown-in-progress');
+
+      // wait for backend response
+      await testSubjects.existOrFail('header-page-supplements > status-badge-in-progress', {
+        timeout: 5000,
+      });
+    },
+
+    /**
+     * Marks a case closed via the status dropdown
+     */
+    async markCaseClosedViaDropdown() {
+      this.openCaseSetStatusDropdown();
+
+      await testSubjects.click('case-view-status-dropdown-closed');
+
+      // wait for backend response
+      await testSubjects.existOrFail('header-page-supplements > status-badge-closed', {
+        timeout: 5000,
+      });
+    },
+
+    /**
+     * Marks a case open via the status dropdown
+     */
+    async markCaseOpenViaDropdown() {
+      this.openCaseSetStatusDropdown();
+
+      await testSubjects.click('case-view-status-dropdown-open');
+
+      // wait for backend response
+      await testSubjects.existOrFail('header-page-supplements > status-badge-open', {
+        timeout: 5000,
+      });
+    },
+
+    async deleteAllBulkListAction() {
+      await testSubjects.setCheckbox('checkboxSelectAll', 'check');
+      const button = await find.byCssSelector('[aria-label="Bulk actions"]');
+      await button.click();
+      await testSubjects.click('cases-bulk-delete-button');
+      await testSubjects.click('confirmModalConfirmButton');
+    },
+
+    async deleteAllCasesFromListUi() {
+      await header.waitUntilLoadingHasFinished();
+      await testSubjects.existOrFail('cases-table', { timeout: 20 * 1000 });
+      let rows: WebElementWrapper[];
+      do {
+        await header.waitUntilLoadingHasFinished();
+        await testSubjects.missingOrFail('cases-table-loading', { timeout: 5000 });
+        rows = await find.allByCssSelector('[data-test-subj*="cases-table-row-"', 100);
+        if (rows.length > 0) {
+          await this.deleteAllBulkListAction();
+          // wait for a second
+          await new Promise((r) => setTimeout(r, 1000));
+          await header.waitUntilLoadingHasFinished();
+        }
+      } while (rows.length > 0);
```

Comment:
Why do we need to delete all cases one by one? Why deleting one is not sufficient?

Reply:
That's the test I decided to  implement. We can also implement another one that deletes only one.